app/src/main/java/com/github/se/orator/ui/profile/StatScreen.kt

```diff
@@ -184,12 +193,12 @@ fun LineChart(xValues: List<Int>, yValues: List<Float>, testTag: String) {
               .height(graphHeight)
               .padding(start = paddingExtraLarge, top = paddingSmall)
               .testTag(testTag)) {
-        val maxX = xValues.maxOrNull()?.toFloat() ?: FULL // full being the const value for 1f
-        val maxY = yValues.maxOrNull() ?: FULL
-        val minY = yValues.minOrNull() ?: ZERO // zero for the value 0f
+        // Use the provided yMin and yMax instead of computing from data
+        val minY = yMin
+        val maxY = yMax
         val yRange = maxY - minY
 
-        // Avoid division by zero: Assign a minimal range if all yValues are the same
+        // If by any chance yRange is 0, fallback to 1f
         val adjustedYRange = if (yRange == ZERO) FULL else yRange
         val xStep = size.width / (xValues.size - 1)
         val yScale = size.height / adjustedYRange
```

Comment:
Personally, I feel the old version was better since it made sure that all values were displayed by dynamically setting maxY based on the graph's data. In the new implementation, maxY is hardcoded, which might cause some values to be cut off. Don't you think it would be better to revert to the old implementation?